skopt/optimizer/base.py

```diff
@@ -170,6 +172,11 @@ def base_minimize(func, dimensions, base_estimator,
         "n_jobs": n_jobs}
     acq_func_kwargs = {"xi": xi, "kappa": kappa}
 
+    if args:
+        opt_func = lambda x_param: func(x_param, args)
```

Comment:
How about using [`partial` from `functools`](https://docs.python.org/3/library/functools.html#functools.partial) instead of our own `lambda` based thing?

Should this be `func(x_param, *args)` instead of `func(x_param, args)`? Also maybe we can just do `func = ...` instead of using `opt_func`?